server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -916,6 +916,9 @@ protected void updateSite2SiteVpnConnectionState(final List<DomainRouterVO> rout
                 }
                 continue;
             }
+            if (router.getIsRedundantRouter() && router.getRedundantState() != RedundantState.MASTER){
```

Comment:
Not per se, as the previous if statement skips the iteration if the router to be checked is not in a running state. And running state is required to do any further state checking.